internal/tool/ssh/deploy.go

```diff
@@ -131,30 +133,38 @@ func deploy(ctx context.Context, args []string) error {
 	}
 }
 
-// copyBinaries copies the tool and the application binary to the given set
-// of locations.
-func copyBinaries(locs []string, dep *protos.Deployment) error {
+// copyBinaries copies the tool and the application binary
+// to the given set of locations. It produces a map which
+// contains the location as a key and the tmpDir in which
+// the binary was copied as a value.
+func copyBinaries(locs []string, dep *protos.Deployment) (map[string]string, error) {
 	ex, err := os.Executable()
 	if err != nil {
-		return err
+		return nil, err
+	}
+
+	tmpDirs, err := getTmpDirs(locs, dep.Id)
+	if err != nil {
+		return nil, err
 	}
 
 	binary := dep.App.Binary
-	remoteDepDir := filepath.Join(os.TempDir(), dep.Id)
-	dep.App.Binary = filepath.Join(remoteDepDir, filepath.Base(dep.App.Binary))
-	for _, loc := range locs {
-		// Make an app deployment directory at each location.
-		cmd := exec.Command("ssh", loc, "mkdir", "-p", remoteDepDir)
+
+	for loc, tmpDir := range tmpDirs {
+		remotePath := filepath.Join(tmpDir, filepath.Base(binary))
+
+		cmd := exec.Command("ssh", loc, "mkdir", "-p", tmpDir)
 		if err := cmd.Run(); err != nil {
-			return fmt.Errorf("unable to create deployment directory at location %s: %w\n", loc, err)
+			return nil, fmt.Errorf("unable to create deployment directory at location %s: %w\n", loc, err)
 		}
 
-		cmd = exec.Command("scp", ex, binary, loc+":"+remoteDepDir)
+		cmd = exec.Command("scp", ex, remotePath, loc+":"+tmpDir)
```

Comment:
this should be `binary`. we copy the binary from the local machine (so the path should be local, as specified in the deployment). Remove `remotePath`, the remote path is simply tmpDir.